### PR TITLE
#4828 - fix tag filter when querying actions

### LINKED ISSUE #4828: Tags filtering is not returning the expected result through the API
**ISSUE TYPE**

Bug Report

**STACKSTORM VERSION**

`st2 3.0dev (367e117), on Python 2.7.12`

**OS / ENVIRONMENT / INSTALL METHOD**

stackstorm-ha helm chart

**SUMMARY**

When trying to filter all the actions based on the tags field an empty list is returned.
For example with the following action definition.

```yaml
---
name: "restart"
runner_type: "local-shell-script"
description: "Restart Gameworld"
enabled: true
entry_point: "webistrano.py"
tags:
  - name: portal-visible
    value: "yes"
parameters:
  gameworld:
    type: "string"
    description: "Enter Gameworld"
    required: true
    position: 0
```

I get this response:

```
GET https://stackstorm/api/v1/actions?tags=portal-visible

HTTP/1.1 200 OK
Server: nginx/1.16.0
Date: Tue, 03 Dec 2019 16:36:29 GMT
Content-Type: application/json
Content-Length: 2
Connection: keep-alive
X-Total-Count: 0

[]

Response code: 200 (OK); Time: 97ms; Content length: 2 bytes
```

### PR DESCRIPTION
**Pull request by winem**
Closes #4828

This PR contains a small fix and does the lookup on the correct key.

Previous implementation returned actions with the name of the provided tag name (value) and not those actions with the specified tag.

### CODE REVIEW

**Comment by winem** [comment]
I saw that there are no tests for the tag filter. 

I will have another look on the tests in the next few days and add them. Will provide the tests in a separate PR if this is merged earlier than that.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -59,7 +59,10 @@ Changed
 
 Fixed
 ~~~~~
+* Fix the action query when filtering tags. The old implementation returned actions which have the
+  provided name as action name and not as tag name. #4828
```
Comment (armab) [inline CHANGELOG.rst]:
```suggestion
  provided name as action name and not as tag name. (bug fix) #4828
```

**Comment by armab** [comment]
@cognifloyd Would be nice to have a second pair of :eyes: on verifying this PR bugfix considering full story in #4828 and more background in 4219.

File: CHANGELOG.rst (hunk 1 above)
Comment (winem) [reply CHANGELOG.rst]: Done.

**Comment by winem** [comment]
Here are 4 curls to show the effect of the fix. 

1. the action itself looks like this:
```
curl -H "X-Auth-Token: xxx" -k https://localhost/api/v1/actions?name=marvel_workflow_demo
[
    {
        "parameters": {
            "environment": {
                "required": true,
                "type": "string"
            },
            "component": {
                "required": true,
                "type": "string"
            },
            "docker_tag": {
                "required": true,
                "type": "string"
            }
        },
        "runner_type": "orquesta",
        "tags": [
            {
                "name": "marvel-project",
                "value": "self-service"
            }
        ],
        "description": "A basic workflow that applies the default workflow for docker image updates.",
        "enabled": true,
        "metadata_file": "actions/marvel_workflow_demo.meta.yaml",
        "entry_point": "workflows/marvel_workflow_demo.yaml",
        "notify": {},
        "output_schema": {},
        "uid": "action:cm_demo_pack:marvel_workflow_demo",
        "pack": "cm_demo_pack",
        "ref": "cm_demo_pack.marvel_workflow_demo",
        "id": "5e4c213b90f52c00e35343ea",
        "name": "marvel_workflow_demo"
    }
]
```

Without the fix a curl to the API with `?tags=marvel-project` did not return any result: 
```
curl -H "X-Auth-Token: xxx" -k https://localhost/api/v1/actions?tags=marvel-project
[]
```

While a curl with the tag provided tag matching the name of the action did return the action:
```
curl -H "X-Auth-Token: xxx" -k 'https://localhost/api/v1/actions?tags=marvel_workflow_demo'
[
    {
        "parameters": {
            "environment": {
                "required": true,
                "type": "string"
            },
            "component": {
                "required": true,
                "type": "string"
            },
            "docker_tag": {
                "required": true,
                "type": "string"
            }
        },
        "runner_type": "orquesta",
        "tags": [
            {
                "name": "marvel-project",
                "value": "self-service"
            }
        ],
        "description": "A basic workflow that applies the default workflow for docker image updates.",
        "enabled": true,
        "metadata_file": "actions/marvel_workflow_demo.meta.yaml",
        "entry_point": "workflows/marvel_workflow_demo.yaml",
        "notify": {},
        "output_schema": {},
        "uid": "action:cm_demo_pack:marvel_workflow_demo",
        "pack": "cm_demo_pack",
        "ref": "cm_demo_pack.marvel_workflow_demo",
        "id": "5e4c213b90f52c00e35343ea",
        "name": "marvel_workflow_demo"
    }
]
```

So, with the fix being applied the query with `?tags=marvel-project` returns the expected action: 
```
curl H "X-Auth-Token: fa589af220454453b245d7f80ee2bbaf" -k 'https://localhost/api/v1/actions?tags=marvel-project'
[
    {
        "parameters": {
            "environment": {
                "required": true,
                "type": "string"
            },
            "component": {
                "required": true,
                "type": "string"
            },
            "docker_tag": {
                "required": true,
                "type": "string"
            }
        },
        "runner_type": "orquesta",
        "tags": [
            {
                "name": "marvel-project",
                "value": "self-service"
            }
        ],
        "description": "A basic workflow that applies the default workflow for docker image updates.",
        "enabled": true,
        "metadata_file": "actions/marvel_workflow_demo.meta.yaml",
        "entry_point": "workflows/marvel_workflow_demo.yaml",
        "notify": {},
        "output_schema": {},
        "uid": "action:cm_demo_pack:marvel_workflow_demo",
        "pack": "cm_demo_pack",
        "ref": "cm_demo_pack.marvel_workflow_demo",
        "id": "5e4c213b90f52c00e35343ea",
        "name": "marvel_workflow_demo"
    }
]
```

**Pull request merged**